src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -14,6 +14,34 @@ if(WITH_MGR_DASHBOARD_FRONTEND)
     add_tox_test(mgr-dashboard-check TOX_ENVS check)
     add_tox_test(mgr-dashboard-openapi TOX_ENVS openapi-check)
   endif()
+
+  # Required for building the dashboard grpc protos at build time
+  set(CEPH_GRPC_VIRTUALENV ${CEPH_BUILD_VIRTUALENV}/ceph-grpc-virtualenv)
+
+  add_custom_command(
+    OUTPUT ${CEPH_GRPC_VIRTUALENV}/lib/python*/site-packages/grpc_tools
+    COMMAND ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=${Python3_EXECUTABLE} ${CEPH_GRPC_VIRTUALENV}
+    COMMAND ${CEPH_GRPC_VIRTUALENV}/bin/pip install grpcio-tools
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMENT "grpcio-tools is being installed")
+
+  # protobuffer files generation
+  add_custom_command(
+    OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/services/proto/gateway_pb2.py" "${CMAKE_CURRENT_SOURCE_DIR}/services/proto/gateway_pb2_grpc.py"
+    COMMAND . ${CEPH_GRPC_VIRTUALENV}/bin/activate && ${CEPH_GRPC_VIRTUALENV}/bin/python -m grpc_tools.protoc --proto_path=. --python_out=. --grpc_python_out=. dashboard/services/proto/gateway.proto && deactivate
```

Comment:
@pecastro thank you for implementing the change to generate pb2 files during build time! Here's a suggestion to make things even more streamlined 😀:

1. remove generated files from version control and outdated copy of the proto file :slightly_smiling_face:
2. update the gateway submodule sha to latest devel  ( see submodule - https://github.com/ceph/ceph/blob/f992060449337cc370ed9bda1e001bf7f4e78040/.gitmodules#L81-L84 ) let me know if you need any help there
3. generate pb2 files in `src/pybind/mgr/dashboard/services/proto` using the protobuf file from the gateway submodule by replacing `dashboard/services/proto/gateway.proto` with `../../../src/nvmeof/gateway/control/proto/gateway.proto`, perhaps?
please share your thoughts